junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/discovery/ClassSelectorResolver.java

```diff
@@ -0,0 +1,133 @@
+			}
+		}
+		else if (isNestedTestClass.test(testClass)) {
+			return toResolution(context.addToParent(() -> selectClass(testClass.getEnclosingClass()),
+				parent -> Optional.of(newNestedClassTestDescriptor(parent, testClass))));
+		}
+		return unresolved();
+	}
+
+	@Override
+	public Resolution resolve(UniqueIdSelector selector, Context context) {
+		UniqueId uniqueId = selector.getUniqueId();
+		UniqueId.Segment lastSegment = uniqueId.getLastSegment();
+		if (ClassTestDescriptor.SEGMENT_TYPE.equals(lastSegment.getType())) {
+			String className = lastSegment.getValue();
+			return ReflectionUtils.tryToLoadClass(className).toOptional().filter(isTestClassWithTests).map(
+				testClass -> toResolution(
+					context.addToParent(parent -> Optional.of(newClassTestDescriptor(parent, testClass))))).orElse(
+						unresolved());
+		}
+		if (NestedClassTestDescriptor.SEGMENT_TYPE.equals(lastSegment.getType())) {
+			String simpleClassName = lastSegment.getValue();
+			return toResolution(context.addToParent(() -> selectUniqueId(uniqueId.removeLastSegment()), parent -> {
+				if (parent instanceof ClassTestDescriptor) {
+					Class<?> parentTestClass = ((ClassTestDescriptor) parent).getTestClass();
+					// TODO add test for resolving unique id of inherited nested test class
+					return ReflectionUtils.findNestedClasses(parentTestClass,
+						isNestedTestClass.and(
+							where(Class::getSimpleName, isEqual(simpleClassName)))).stream().findFirst().flatMap(
+								testClass -> Optional.of(newNestedClassTestDescriptor(parent, testClass)));
+				}
+				return Optional.empty();
+			}));
+		}
+		return unresolved();
+	}
+
+	private ClassTestDescriptor newClassTestDescriptor(TestDescriptor parent, Class<?> testClass) {
+		return new ClassTestDescriptor(
+			parent.getUniqueId().append(ClassTestDescriptor.SEGMENT_TYPE, testClass.getName()), testClass,
+			configuration);
+	}
+
+	private NestedClassTestDescriptor newNestedClassTestDescriptor(TestDescriptor parent, Class<?> testClass) {
+		return new NestedClassTestDescriptor(
+			parent.getUniqueId().append(NestedClassTestDescriptor.SEGMENT_TYPE, testClass.getSimpleName()), testClass,
+			configuration);
+	}
+
+	private Resolution toResolution(Optional<ClassTestDescriptor> testDescriptor) {
+		return testDescriptor.map(it -> {
+			Class<?> testClass = it.getTestClass();
+			// @formatter:off
+            return Resolution.match(Match.exact(it, () -> {
+                Stream<MethodSelector> methods = findMethods(testClass, isTestOrTestFactoryOrTestTemplateMethod).stream()
+                        .map(method -> selectMethod(testClass, method));
+				Stream<ClassSelector> nestedClasses = findNestedClasses(testClass, isNestedTestClass).stream()
+						.map(DiscoverySelectors::selectClass);
+				return Stream.concat(methods, nestedClasses).collect(toCollection((Supplier<Set<DiscoverySelector>>) LinkedHashSet::new));
+            }));
+            // @formatter:on
```

Comment:
@marcphilipp It looks to me that the code within this `@formatter:(off|on)` block has some space/tab indenting issues. :)

Reply:
/me mumbles _Turning formatters off in 2019..._ 😉

Reply:
@sormuras Indeed. :wink: 

But on a more serious note, it makes me wonder if Eclipse has a setting that would allow method chains to be formatted with one method per line, without the JUnit 5 team having to manually use `@formatter(off|on)`... 🤔

(Am I right to think that using google-java-format, which IIUC would solve this problem more easily, was considered by the JUnit 5 team but decided against?)

Reply:
> ... using google-java-format ... was considered by the JUnit 5 team but decided against?

True. But I'm only allowed to bring up that topic once a year. Mentioned it last week already -- so when is 2020?! 😈

Reply:
@jbduncan I don't see any space/tab indenting issues. What am I missing?

Reply:
![image](https://user-images.githubusercontent.com/2319838/53984225-3757ce80-4119-11e9-992d-7204012b856b.png)

Reply:
Could be an GitHub display only issue. Didn't check the whitespace chars in my IDE...

Reply:
Indeed it could be just a GitHub issue; I don't have an IDE with this PR checked out to confirm. :sweat_smile:

Reply:
Nope: not a GitHub. They were spaces instead of tabs. Fixed here: f114fd009778bb4a33cda97a42bdf68d71348221 😉

Reply:
Haha, thank you for confirming and cleaning it up, @sbrannen. :smile: